README.md

```diff
@@ -1,3 +1,270 @@
+	// will only be populated from command line flags
+	Val3 bool `dialsflag:"val-3"`
+}
+
+func main() {
+	config := &Config{
+		// Val1 has a default value of "hello" and will be overwritten by the
+		// sources if there is a corresponding field for Val1
+		Val1: "hello",
+	}
+
+	// To read from other source files such as JSON, and TOML, use
+	// "&json.Decoder{}" or "&toml.Decoder{}"
+	fileSrc, fileErr := file.NewSource("path/to/config", &yaml.Decoder{})
+	if fileErr != nil {
+		// error handling
+	}
+
+	// Define a `dials.Source` for command line flags. Consider using the dials pflag library
+	// if the application uses the spf13/pflag package
+	flagSet, flagErr := flag.NewCmdLineSet(flag.DefaultFlagNameConfig(), config)
+	if flagErr != nil {
+		// error handling
+	}
+
+	// use the environment source to get values from environment variables
+	envSrc := &env.Source{}
+
+	// the Config struct will be populated in the order in which the sources are
+	// passed in the Config function with increasing precedence. So the fileSrc value
+	// will overwrite the flagSet value if they both were to set for the same field
+	d, err := dials.Config(context.Background(), config, envSrc, flagSet, fileSrc)
+	if err != nil {
+		// error handling
+	}
+
+	// Fill populates the config struct
+	c := &Config{}
+	d.Fill(c)
+	fmt.Printf("Config: %+v\n", c)
+}
+```
+
+If the above code is run with the following YAML file (make sure to change the path in the code):
+
+``` yaml
+b: valueb
+val-3: false
+```
+
+and the following commands 
+
+``` 
+export VAL_2=5
+go run main.go --val-3
+```
+ 
+the output will be `Config: &{Val1:valueb Val2:5 Val3:true}`. Note that even when val-3 is defined in the yaml file and the file source takes precedence, 
+only the value from command line flag populates the config due to the special `dialsflag` tag. If you prefer the value for `Val3` be overwritten by other sources, then
+use the `dials` tag instead of the `dialsflag` tag
+
+
+
+### Watching file source
+If you wish to watch the config file and make updates to your configuration, use the watching source. This functionality is available in the `ez` package by passing the `true` boolean to the functions. The `WatchingSource` can be used when you want to further customize the configuration as well. Please note that the Watcher interface is likely to change in the near future.
+
+``` go
+	// NewWatchSource also has watch options that can be passed to have the
+	// ability to use a ticker for polling, set a logger, and more
+	watchingFileSource, fsErr := file.NewWatchingSource(
+		"path/to/config", &yaml.Decoder{})
+
+	if fsErr != nil {
+		// error handling
+		return
+	}
+
+	// additional sources can be passed along with the watching file source and the
+	// precedence order will still be dictated by the order in which the sources are
+	// defined in the Config function.
+	d, err := dials.Config(context.Background(), config, watchingFileSource)
```

Comment:
Let's explicitly call out using a non-background context here and include a context coming from `context.WithCancel`.